### DISCUSSION TITLE
invalid value obtained from hashtable

### DISCUSSION THREAD

**Discussion by commonuserlol** [vfsfitvnm/frida-il2cpp-bridge#280] [Q&A]
hello, i have class `ExitGames.Client.Photon.Hashtable` which extends `System.Collections.Generic.Dictionary<System.Object,System.Object>`, here is its dump:

```
class ExitGames.Client.Photon.Hashtable : System.Collections.Generic.Dictionary<System.Object,System.Object>
{
    
    System.Void .ctor(); // 0x024473d4
    System.Void .ctor(System.Int32 x); // 0x02447428
    System.Object get_Item(System.Object key); // 0x02447484
    System.Void set_Item(System.Object key, System.Object value); // 0x024474f8
    System.Collections.Generic.IEnumerator<System.Collections.DictionaryEntry> GetEnumerator(); // 0x0244755c
    System.String ToString(); // 0x02447684
}
```
there is also a method that receives one of the instances of this class. when i print it via `console.log` everything is fine (output: `(System.String)RoundStartTime=(System.Double)23389.599, (System.String)Time=(System.Double)23389.599, (System.String)C2=(System.Byte)22, (System.String)bomberId=(System.Int32)13, (System.String)Round=(System.Int32)3, (System.String)RouteToGameStateId=(System.Byte)0`)
i'm interested in the value of the "Round".
`const round = hashtable.method("get_Item").invoke<number>(Il2Cpp.String.from("Round"));` but instead of the real value i get `-479416352` (no matter what the real value).
does anyone know why this is happening?

**Comment by ExternalAddress4401** [comment]
I see this is two weeks old but have you tried with the other values?

`const round = hashtable.method("get_Item").invoke<number>(Il2Cpp.String.from("RoundStartTime")); console.log(round);`

If that also returns a strange incorrect looking value then I feel like you might need to be print `console.log(round.value)` instead.
